celery/canvas.py

```diff
@@ -1170,21 +1170,29 @@ def _apply_tasks(self, tasks, producer=None, app=None, p=None,
             # we are able to tell when we are at the end by checking if
             # next_task is None.  This enables us to set the chord size
             # without burning through the entire generator.  See #3021.
+            group_ids = set()
```

Comment:
If this set is used only for the sanity check we should avoid using it if `__debug__` is False.